File: backend/app/models/appointment.py
import enum
import uuid
from datetime import datetime

from sqlalchemy import String, Boolean, DateTime, Integer, Enum, ForeignKey, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base


class RiskLevel(str, enum.Enum):
    LOW = "low"
    MODERATE = "moderate"
    HIGH = "high"
    CRITICAL = "critical"


class AppointmentStatus(str, enum.Enum):
    SCHEDULED = "scheduled"
    CONFIRMED = "confirmed"
    CHECKED_IN = "checked_in"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    CANCELLED = "cancelled"
    NO_SHOW = "no_show"


class Appointment(Base):
    __tablename__ = "appointments"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    patient_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("patients.id")
    )
    title: Mapped[str] = mapped_column(String(200))
    description: Mapped[str] = mapped_column(String, nullable=True)
    risk_level: Mapped[RiskLevel] = mapped_column(
        Enum(RiskLevel, values_callable=lambda e: [s.value for s in e]),
        default=RiskLevel.LOW,
    )
    appointment_date: Mapped[datetime] = mapped_column(DateTime)
    duration_minutes: Mapped[int] = mapped_column(Integer, default=30)
    provider: Mapped[str] = mapped_column(String(200))
    status: Mapped[AppointmentStatus] = mapped_column(
        Enum(AppointmentStatus, values_callable=lambda e: [s.value for s in e]),
        default=AppointmentStatus.SCHEDULED,
    )
    location: Mapped[str] = mapped_column(String(200), nullable=True)
    notes: Mapped[str] = mapped_column(String, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), onupdate=func.now()
    )

    patient = relationship("Patient", backref="appointments")

File: frontend/src/types/appointment.ts
export type RiskLevel = "low" | "moderate" | "high" | "critical";

export type AppointmentStatus =
  | "scheduled"
  | "confirmed"
  | "checked_in"
  | "in_progress"
  | "completed"
  | "cancelled"
  | "no_show";

export interface Appointment {
  id: string;
  patient_id: string;
  title: string;
  description: string | null;
  risk_level: RiskLevel;
  appointment_date: string;
  duration_minutes: number;
  provider: string;
  status: AppointmentStatus;
  location: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface AppointmentCreatePayload {
  patient_id: string;
  title: string;
  description?: string | null;
  risk_level?: RiskLevel;
  appointment_date: string;
  duration_minutes?: number;
  provider: string;
  location?: string | null;
  notes?: string | null;
}

export interface AppointmentUpdatePayload {
  title?: string | null;
  description?: string | null;
  risk_level?: RiskLevel | null;
  appointment_date?: string | null;
  duration_minutes?: number | null;
  provider?: string | null;
  status?: AppointmentStatus | null;
  location?: string | null;
  notes?: string | null;
}
